datadog_checks_base/datadog_checks/base/utils/http.py

```diff
@@ -170,7 +171,16 @@ def __init__(self, instance, init_config, remapper=None, logger=None):
         auth = None
         if config['password']:
             if config['username']:
-                auth = (config['username'], config['password'])
+                if config['auth_type']:
```

Comment:
Suggestion using my other comments:

```suggestion
                auth_type = config.get('auth_type', 'basic').lowercase():
                if auth_type == 'digest':
                    auth = requests.auth.HTTPDigestAuth(config['username'], config['password'])
                else:
                    auth = requests.auth.HTTPBasicAuth(config['username'], config['password'])
```

Reply:
is it ok to default to 'basic' if they input other kinds of `auth_type` such as `oauth`?

Reply:
We should probably log a message like "auth_type <XXX> is not supported, defaulting to basic auth".

Reply:
going to make changes locally, removing the `auth = (config['username'], config['password'])` part breaks the `test_config_basic` test so I'm going to modify it and push up

Reply:
We have extra logic to differenciate with HTTPBasicAuth and (username, password). Can we simplify to this then?:
```suggestion
                auth_type = config.get('auth_type', 'basic').lowercase():
                if auth_type == 'digest':
                    auth = requests.auth.HTTPDigestAuth(config['username'], config['password'])
                else:
                    if auth_type != 'basic':
                        self.logger.warning('auth_type {} is not supported, defaulting to basic', auth_type)
                    auth = (config['username'], config['password'])
```

This will have the benefits explained earlier and would also not break tests.